File: studio/src/design/consumer-mode.ts
export type ProjectConsumerMode = 'agent_anip' | 'hybrid'

export const DEFAULT_CONSUMER_MODE: ProjectConsumerMode = 'hybrid'

export const CONSUMER_LABEL_PREFIX = 'consumer:'

const CONSUMER_MODE_LABELS: Record<ProjectConsumerMode, string> = {
  hybrid: 'Both people and agents',
  agent_anip: 'Agents/tools through ANIP',
}

export const CONSUMER_MODE_OPTIONS: Array<{
  value: ProjectConsumerMode
  label: string
  description: string
}> = [
  {
    value: 'hybrid',
    label: CONSUMER_MODE_LABELS.hybrid,
    description: 'Design both the human product workflow and the governed ANIP capability surface.',
  },
  {
    value: 'agent_anip',
    label: CONSUMER_MODE_LABELS.agent_anip,
    description: 'Optimize the design for direct agent/tool consumption through explicit ANIP capabilities.',
  },
]

export function consumerModeTag(mode: ProjectConsumerMode): string {
  return `${CONSUMER_LABEL_PREFIX}${mode}`
}

export function labelsWithConsumerMode(labels: string[] | undefined, mode: ProjectConsumerMode): string[] {
  const preserved = (labels ?? []).filter(label => !label.startsWith(CONSUMER_LABEL_PREFIX))
  return [...preserved, consumerModeTag(mode)]
}

export function consumerModeFromLabels(labels: string[] | undefined): ProjectConsumerMode {
  const match = (labels ?? []).find(label => label.startsWith(CONSUMER_LABEL_PREFIX))
  const value = match?.slice(CONSUMER_LABEL_PREFIX.length)
  if (value === 'agent_anip' || value === 'hybrid') return value
  return DEFAULT_CONSUMER_MODE
}

export function consumerModeLabel(mode: ProjectConsumerMode): string {
  return CONSUMER_MODE_LABELS[mode] ?? CONSUMER_MODE_LABELS[DEFAULT_CONSUMER_MODE]
}

File: studio/server/consumer_mode.py
"""Helpers for Studio project consumer-mode labels."""

from __future__ import annotations

from typing import Iterable


DEFAULT_CONSUMER_MODE = "hybrid"
CONSUMER_LABEL_PREFIX = "consumer:"

CONSUMER_MODE_LABELS = {
    "human_app": "People through an app",
    "agent_anip": "Agents and tools through ANIP",
    "hybrid": "Both people and agents",
}


def normalize_consumer_mode(value: str | None) -> str:
    cleaned = (value or "").strip().lower()
    if cleaned in CONSUMER_MODE_LABELS:
        return cleaned
    return DEFAULT_CONSUMER_MODE


def consumer_mode_from_labels(labels: Iterable[str] | None) -> str:
    for label in labels or []:
        if isinstance(label, str) and label.startswith(CONSUMER_LABEL_PREFIX):
            return normalize_consumer_mode(label[len(CONSUMER_LABEL_PREFIX) :])
    return DEFAULT_CONSUMER_MODE


def consumer_mode_label(mode: str | None) -> str:
    normalized = normalize_consumer_mode(mode)
    return CONSUMER_MODE_LABELS[normalized]


def consumer_mode_tag(mode: str | None) -> str:
    return f"{CONSUMER_LABEL_PREFIX}{normalize_consumer_mode(mode)}"
